src/equations/ideal_glm_mhd_multiion_3d.jl

```diff
@@ -1094,4 +1471,46 @@ end
 
     return c_f
 end
+
+@inline function calc_fast_wavespeed(cons, normal_direction::AbstractVector,
+                                     equations::IdealGlmMhdMultiIonEquations3D)
+    B1, B2, B3 = magnetic_field(cons, equations)
+    psi = divergence_cleaning_field(cons, equations)
+
+    norm_squared = (normal_direction[1] * normal_direction[1] +
+                    normal_direction[2] * normal_direction[2] +
+                    normal_direction[3] * normal_direction[3])
+
+    c_f = zero(real(equations))
+    for k in eachcomponent(equations)
+        rho, rho_v1, rho_v2, rho_v3, rho_e = get_component(k, cons, equations)
+
+        rho_inv = 1 / rho
+        v1 = rho_v1 * rho_inv
+        v2 = rho_v2 * rho_inv
+        v3 = rho_v3 * rho_inv
+        gamma = equations.gammas[k]
+        p = (gamma - 1) *
+            (rho_e - 0.5f0 * rho * (v1^2 + v2^2 + v3^2) - 0.5f0 * (B1^2 + B2^2 + B3^2) -
+             0.5f0 * psi^2)
+        a_square = gamma * p * rho_inv
+        inv_sqrt_rho = 1 / sqrt(rho)
+
+        b1 = B1 * inv_sqrt_rho
+        b2 = B2 * inv_sqrt_rho
+        b3 = B3 * inv_sqrt_rho
+        b_square = b1^2 + b2^2 + b3^2
+        b_dot_n_squared = (b1 * normal_direction[1] +
+                           b2 * normal_direction[2] +
+                           b3 * normal_direction[3])^2 / norm_squared
+
+        c_f = max(c_f,
+                  sqrt((0.5f0 * (a_square + b_square) +
+                        0.5f0 *
+                        sqrt((a_square + b_square)^2 - 4 * a_square * b_dot_n_squared)) *
```

Comment:
```suggestion
                        0.5f0 * sqrt((a_square + b_square)^2 - 
                        4 * a_square * b_dot_n_squared)) *
```

Maybe this is nicer formatting?

Reply:
`4 * a_square * b_dot_n_squared` is inside `sqrt`. Do you mean this?
```suggestion
                        0.5f0 * sqrt((a_square + b_square)^2 - 
                                     4 * a_square * b_dot_n_squared)) *
```

Reply:
It seems that none of the two suggestions was. I found a third alternative 3bf6ccb8bcbd4f6df746d8027d06d36dd1fa17fc